test/scenarios.py

```diff
@@ -7875,4 +7875,52 @@
             ),
         },
     ),
+    Scenario(
+        "scenario39",
+        "Produce a situation in which the receiver has stalled transfer, but sender does not have it anymore. Expect receiver to check the transfer state eventually",
+        {
+            "ren": ActionList(
+                [
+                    action.Start("172.20.0.5"),
+                    action.NewTransfer("172.20.0.15", ["/tmp/testfile-small"]),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.SleepMs(200),
+                    action.Stop(),
+                    action.Start("172.20.0.5"),
+                    action.Sleep(61),  # 60s is a check request interval
+                    action.Stop(),
+                ]
+            ),
+            "stimpy": ActionList(
+                [
+                    action.Start("172.20.0.15"),
+                    action.Wait(
+                        event.Receive(
+                            0,
+                            "172.20.0.5",
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.ExpectCancel([0], True),
+                ]
```

Comment:
Additional test: storage API with the stale transfer on both ends

Reply:
Yeah, could be even embedded in this test

Reply:
done